engine/tests/persistence/test_serialization.py

```
@@ -32,3 +34,19 @@ def test_round_trip(serializer):
 
     # Assertion to verify that the deserialized data is equivalent to the original
     assert unflat == unstructured
+
+
+def test_json_bytes_round_trip_uses_explicit_tagged_shape() -> None:
+    payload = {"blob": b"\x12\x34\xab\xcd"}
+
+    flat = JsonSerializationHandler.serialize(payload)
```

Ruff (0.15.6)

[error] `JsonSerializationHandler` may be undefined, or defined from star imports

(F405)